src/components/Categories/CategoriesTable.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from "react";
+import Paper from "@material-ui/core/Paper";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableContainer from "@material-ui/core/TableContainer";
+import TableHead from "@material-ui/core/TableHead";
+import TablePagination from "@material-ui/core/TablePagination";
+import TableRow from "@material-ui/core/TableRow";
+import { withStyles, WithStyles } from "@material-ui/core/styles";
+import { createStyles, Theme } from "@material-ui/core/styles";
+import { Actions } from "../../types";
+import { TableContent } from "../../types";
+import EditIcon from "@material-ui/icons/Edit";
+import DeleteForeverIcon from "@material-ui/icons/DeleteForever";
+import IconButton from "@material-ui/core/IconButton";
+import DeleteCategoryDialog from "./DeleteCategoryDialog";
+import CategoryFormDialog from "./CategoryFormDialog";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    root: {
+      width: "75%",
+    },
+    container: {
+      maxHeight: 300,
+    },
+    tableCell: {
+      minWidth: 170,
+      align: "left",
+    },
+  });
+
+const CategoriesTable: React.FC<
+  TableContent & Actions & WithStyles<typeof styles>
+> = (props) => {
+  const classes = props.classes;
+  const handleDelete = props.onDelete;
+  const incomingRows = props.rows;
+  const columns = props.columns;
+  const handleEdit = props.onEdit;
+
+  const [rows, setRows] = useState(incomingRows);
+  useEffect(() => {
+    setRows(incomingRows);
+  }, [incomingRows]);
```

Comment:
in such cases it is super better to use  useMemo over useState+ useEffect ... so these will be 
const rows = useMemo(()=>incomingRows.slice(),[incomingRows]);

... and if you moved this down you might move the rows.slice to this memorized function too.